### PR TITLE
Added all combat wrapper classes

### PR DESCRIPTION
**Pull request by NoJokeFNA**
https://wiki.vg/Protocol#Enter_Combat_Event
https://wiki.vg/Protocol#End_Combat_Event
https://wiki.vg/Protocol#Death_Combat_Event
https://wiki.vg/index.php?title=Protocol&oldid=14446#Combat_Event
Tested Server-Versions: 1.8.8, 1.12.2, 1.13.2, 1.14.4, 1.16.5, 1.18.2

### CODE REVIEW

File: api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerDeathCombatEvent.java [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+        this.playerId = playerId;
+        this.entityId = entityId;
+        this.deathMessage = deathMessage;
+    }
+
+    @Override
+    public void read() {
+        this.playerId = readVarInt();
+        this.entityId = readInt();
+        this.deathMessage = readString();
```
Comment (MWHunter) [inline api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerDeathCombatEvent.java:45]: This is a Chat (Component datatype) not a string

File: api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerCombatEvent.java [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+        this.combat = Combat.VALUES[readVarInt()];
+        switch (combat) {
+            case END_COMBAT:
+                this.duration = readVarInt();
+                this.entityId = readInt();
+                break;
+            case ENTITY_DEAD:
+                this.playerId = readVarInt();
+                this.entityId = readInt();
+                this.deathMessage = readString();
```
Comment (MWHunter) [inline api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerCombatEvent.java:71]: This changes cross version

**Review by MWHunter — CHANGES_REQUESTED** [review]

File: api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerCombatEvent.java (hunk 2 above)
Comment (NoJokeFNA) [reply api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerCombatEvent.java:71]: In which ones?

Comment (NoJokeFNA) [reply api/src/main/java/com/github/retrooper/packetevents/wrapper/play/server/WrapperPlayServerCombatEvent.java]: Honestly, I can't find any changes cross versions

**Pull request merged**